src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialsStore.java

```diff
@@ -44,9 +47,9 @@ public boolean hasPermission(@NonNull Authentication authentication, @NonNull Pe
     @NonNull
     @Override
     public List<Credentials> getCredentials(@NonNull Domain domain) {
-        // Only the global domain is supported
-        if (Domain.global().equals(domain) && Jenkins.getInstance().hasPermission(CredentialsProvider.VIEW))
-            return provider.getCredentials(Credentials.class, Jenkins.getInstance(), ACL.SYSTEM);
+        if(Jenkins.getInstance().hasPermission(CredentialsProvider.VIEW)) {
```

Comment:
the Access check should be scoped to the context.  if you have CredentialsProvider.View on a folder (but not Jenkins itself) you should be able to view credentials assigned to that folder (or a sub folder).

Additionally I beleive we still only support a global domain?

```suggestion
        // Only the global domain is supported
        if (Domain.global().equals(domain)) {
            // walk the parent until we get the AccessControlled object
            AccessControlled ac = null;
            ItemGroup ig = context;
            while (ac == null) {
                if (ig instanceOf AccessControlled) {
                    ac = (AccessControlled)ig;
                } else {
                    ig = ((Item) ig).getParent();
                }         
            }
            if (ac.hasPermission(CredentialsProvider.VIEW) {
                return provider.getCredentials(Credentials.class, context, ACL.SYSTEM);
            }
            return Collections.emptyList();
        }
```